packages/block-library/src/search/edit.native.js

```diff
@@ -72,19 +113,81 @@ export default function SearchEdit( { attributes, setAttributes } ) {
 		</BlockControls>
 	);
 
+	const renderTextField = () => {
+		return (
+			<TextInput
+				className="wp-block-search__input"
+				style={ styles.searchTextInput }
+				label={ null }
+				value={ placeholder }
+				placeholder={
+					placeholder ? undefined : __( 'Optional placeholder…' )
+				}
+				onChangeText={ ( newVal ) =>
+					setAttributes( { placeholder: newVal } )
+				}
+			/>
+		);
+	};
+
+	const renderButton = () => {
+		return (
+			<View style={ styles.buttonContainer }>
+				{ buttonUseIcon && (
+					<Button
+						className="wp-block-search__button"
+						icon={ search }
+					/>
+				) }
+
+				{ ! buttonUseIcon && (
+					<RichText
+						className="wp-block-search__button"
+						style={ richTextStyles.searchButton }
+						placeholder={ __( 'Add button text' ) }
+						value={ buttonText }
+						identifier="text"
+						withoutInteractiveFormatting
+						onChange={ ( html ) =>
+							setAttributes( { buttonText: html } )
+						}
+						minWidth={ MIN_BUTTON_WIDTH }
+						textAlign="center"
+					/>
+				) }
+			</View>
+		);
+	};
+
 	return (
-		<View>
+		<View { ...blockProps } style={ styles.searchBlockContainer }>
 			{ controls }
 
 			{ showLabel && (
 				<RichText
+					className="wp-block-search__label"
+					style={ {
+						...styles.searchLabel,
```

Comment:
Nice use of [spread attributes](https://reactjs.org/docs/jsx-in-depth.html#spread-attributes) 👍